[docs] new marketplace docs (#6057)
ToolJet/ToolJet: docs/docs/contributing-guide/marketplace/marketplace-setup.md

---
id: marketplace-setup
title:  'Marketplace: Setup'
---

> The Marketplace offers custom plugins that can be installed in your ToolJet workspace. This guide aims to assist you in creating a new plugin for the ToolJet marketplace.

Author: ```suggestion
The Marketplace offers custom plugins that can be installed in your ToolJet instance. This guide aims to assist you in creating a new plugin for the ToolJet marketplace.
```

## Requirements
- [Node.js](https://nodejs.org/en/download/) **(v18.3.0)**
- [npm](https://www.npmjs.com/get-npm) **(v8.11.0)**

## Getting started

### Step 1. Enabling the marketplace for your instance

> In order to **activate** the marketplace for your ToolJet instance, you must set the following evironment variables in your **`.env`** file:

Author: ```suggestion
To enable the marketplace for your ToolJet instance, you need to specify the following environment variables in your **`.env`** file:
```

#### Marketplace feature enable

Use this environment variable to enable/disable the feature that allows users to use the [marketplace](/docs/marketplace).

| variable                   | value             |
| -------------------------- | ----------------- |
| ENABLE_MARKETPLACE_FEATURE | `true` or `false` |

#### Enable Marketplace plugin developement mode

> Use this environment variable to enable/disable the developement mode that allows developers to build the plugin.

Author: ```suggestion
The use of this environment variable facilitates plugin development by enabling automatic builds whenever package changes occur, thus simplifying the development process. Moreover, it also incorporates a reload button that retrieves all the recent local modifications from the file system for installed plugins, making it a valuable feature for improving the overall development experience.
```

> Use this environment variable to enable/disable the developement mode that allows developers to build the plugin.

Reviewer: Can re-write to something similar, 'This environment variable enables developers to build plugins with greater ease by allowing for auto builds when there are changes in the packages. Additionally, a reload button is included for installed plugins to retrieve all the latest local changes from the file system. This feature is particularly useful for enhancing the development experience.'

| variable                   | value             |
| -------------------------- | ----------------- |
| ENABLE_MARKETPLACE_DEV_MODE | `true` or `false` |

Please note that the marketplace is not enabled by default. After updating the variable, restart your ToolJet instance. 

For information on running ToolJet on your local machine, please refer to the instructions provided **[here](/docs/category/contributing-guide)**. You can access the marketplace by navigating to the **'/integrations'** route.

### Step 2: Installation of tooljet-cli

> To create and publish plugins for the ToolJet marketplace, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:

Reviewer: ```suggestion
In order to manage plugins for the ToolJet marketplace, including creating, updating, and deleting them,....
```

> To create and publish plugins for the ToolJet marketplace, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:

Author: ```suggestion
In order to manage plugins for the ToolJet marketplace, including creating, updating, and deleting, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:
```

```bash
npm install -g tooljet-cli

# Ensure the installation was successful
tooljet --version
```

> **Now that we have finished setting up the environment for Marketplace Developer mode, we can move on to the next section and begin [developing the first plugin](/docs/contributing-guide/marketplace/creating-a-plugin).**

Reviewer: ```suggestion
Having completed the environment setup for Marketplace Developer mode, we can proceed to the next section and commence with [developing the first plugin](https://chat.openai.com/docs/contributing-guide/marketplace/creating-a-plugin).
```

---
id: marketplace-setup
title:  'Marketplace: Setup'
---

> The Marketplace offers custom plugins that can be installed in your ToolJet instance. This guide aims to assist you in creating a new plugin for the ToolJet marketplace.

Reviewer: It may be more appropriate to use a title such as "Marketplace Development Setup" for a guide that covers the initial setup steps for developing plugins, rather than simply "Marketplace: Setup".

| variable                   | value             |
| -------------------------- | ----------------- |
| ENABLE_MARKETPLACE_DEV_MODE | `true` or `false` |

Please note that the marketplace is not enabled by default. After updating the variable, restart your ToolJet instance. 

For information on running ToolJet on your local machine, please refer to the instructions provided **[here](/docs/category/contributing-guide)**. You can access the marketplace by navigating to the **'/integrations'** route.

### Step 3: Installation of tooljet-cli

In order to manage plugins for the ToolJet marketplace, including creating, updating, and deleting, you will need to utilize **[tooljet-cli](https://www.npmjs.com/package/@tooljet/cli)**. This can be installed via npm by entering the following command:
```bash
npm install -g tooljet-cli

# Ensure the installation was successful
tooljet --version
```

> Having completed the environment setup for Marketplace Developer mode, we can proceed to the next section and commence with [developing the first plugin](https://chat.openai.com/docs/contributing-guide/marketplace/creating-a-plugin).

Reviewer: I believe the URL may be incorrect.

> Having completed the environment setup for Marketplace Developer mode, we can proceed to the next section and commence with [developing the first plugin](https://chat.openai.com/docs/contributing-guide/marketplace/creating-a-plugin).

Author: ```suggestion
Having completed the environment setup for Marketplace Developer mode, we can proceed to the next section and commence with [developing the first plugin](/docs/contributing-guide/marketplace/creating-a-plugin).
```
